File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque


def input():
    return sys.stdin.readline().rstrip()


n = int(input())
queue = deque([])
command_dict = {
    'push_front': queue.appendleft,
    'push_back': queue.append,
    'pop_front': lambda: -1 if not queue else queue.popleft(),
    'pop_back': lambda: -1 if not queue else queue.pop(),
    'size': lambda: len(queue),
    'empty': lambda: 1 if not queue else 0,
    'front': lambda: -1 if not queue else queue[0],
    'back': lambda: -1 if not queue else queue[-1],
}
for _ in range(n):
    commands = input().split()
    cmd = commands[0]
    if cmd in ['push_front', 'push_back']:
        command_dict[cmd](commands[1])
    else:
        print(command_dict[cmd]())

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <string>
#include <deque>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n, x;
	string cmd;
	deque<int> dq;
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd == "push_front") {
			cin >> x;
			dq.push_front(x);
			continue;
		}
		else if (cmd == "push_back") {
			cin >> x;
			dq.push_back(x);
			continue;
		}
		else if (cmd == "pop_front") {
			if (dq.empty()) cout << -1;
			else {
				cout << dq.front();
				dq.pop_front();
			}
		}
		else if (cmd == "pop_back") {
			if (dq.empty()) cout << -1;
			else {
				cout << dq.back();
				dq.pop_back();
			}
		}
		else if (cmd == "size") {
			cout << dq.size();
		}
		else if (cmd == "empty") {
			cout << dq.empty() ? 1 : 0;
		}
		else if (cmd == "front") {
			if (dq.empty()) cout << -1;
			else cout << dq.front();
		}
		else if (cmd == "back") {
			if (dq.empty()) cout << -1;
			else cout << dq.back();
		}
		cout << '\n';
	}
}